src/ParameterPostProcessor.php

```diff
@@ -60,10 +55,22 @@ private function convertValues(array $values, string $prefix = '') : array
 
             $convertedValues[$prefix . $key] = $value;
             if (is_array($value)) {
-                $convertedValues += $this->convertValues($value, $prefix . $key . '.');
+                $convertedValues += $this->resolveNestedParameters($value, $prefix . $key . '.');
             }
         }
 
         return $convertedValues;
     }
+
+    private function getResolvedParameters() : ParameterBag
+    {
+        $resolved = $this->resolveNestedParameters($this->parameters);
+        $bag = new ParameterBag($resolved);
+        try {
+            $bag->resolve();
+            return $bag;
+        } catch (SymfonyParameterNotFoundException $exception) {
```

Comment:
What if we remove the `try...catch` here and move the execution of this method in `__invoke` into `try...catch` we have there?

Reply:
We cannot use `resolve` before we resolved the recursive parameters as they may miss.
This would throw an exception even if the parameter will exist (its not yet resolved yet).

The process is as follows:

1. Injecting parameters
2. Resolve recursive parameters like foo.bar (they are usually written as `['foo' => ['bar' => 'baz']]`)
3. Re-resolve again to resolve parameters which may use `foo.bar` itself (`['bar' => ['baz' => '%foo.bar%']`)
4. Resolve the configuration itself

Reply:
Sorry, maybe I was not clear, this is not what I meant. See my comment here:
https://github.com/zendframework/zend-config-aggregator-parameters/pull/1/files#r247840650

Reply:
I still dont really get what you mean.
Can you paste an example in your next comment?

Reply:
Oh okay, so you want me to catch within `__invoke` instead in the private method.

Again, is there a reason why I should do this? I only would like to catch code which throws exceptions and not code which does not. Would love to understand that.

Reply:
```php
public function __invoke(array $config) : array
{
    try {
        // call of below method is inside try...catch not before, as it is now.
        $parameters = $this->getResolvedParameters();

        array_walk_recursive($config, function (&$value) use ($parameters) {
            $value = $parameters->unescapeValue($parameters->resolveValue($value));
        });
    } catch (SymfonyParameterNotFoundException $exception) {
        throw ParameterNotFoundException::fromException($exception);
    }

    $config['parameters'] = $parameters->all();
    return $config;
}
```

and remove `try...catch` in `getResolvedParameters`.

Reply:
You catch the same exception twice and re-throw it in the same style, for me looks like it should be handled by one try...catch instruction.